### ISSUE TITLE
When are raw training data made publicly availailable ?

### ISSUE THREAD

**Issue by Ishinoshita** [gcp/leela-zero#1916]
Hi,
Is there a fix schedule/rule for the availability of raw training data on http://leela.online-go.com/training/ (anything automated) or does it purely rely on manual operation, when devs have bandwidth ? Just curious to know (question prompted by issues like [this one](https://github.com/gcp/leela-zero/issues/1906)).
Thanks !

**Comment by bjiyxo** [comment]
It is manual uploaded by @gcp once a week.

**Comment by gcp** [comment]
One of the last things I need to automate at some point...

**Comment by Ishinoshita** [comment]
@gcp Having the training data more frequently would help doing sanity checks both on sgf and at data level, when someone claim there are weird games uploaded. 

Such claims sometime seem highly plausible, sometime just reflect misunderstanding of how randomness works. Anyway, they are difficult to decide on on the sole basis of the online latest self-play games sgf.

The supposedly 'weird' games should be found in the last generated training chunks, a priori. Trying to retrieve them 15 days later among the full chunks list is like looking for a needle in a hey stack. But even with a daily dump, that might not suffice to narrow down the search, as training data are shared only once the network has been demoted. So, in most cases, the traing data of today's last 100 self-play games will not be published before days. 

But, to start with, do you think there is any usefulness at all in trying to do the kind of analysis I did recently? If not really, then the point here above is void of course.

**Comment by alreadydone** [comment]
To find a game in the training trunks, I think you can loadsgf and then dump_supervised and pick any hexadecimal string near the end of the generated training data file. Then you need to search for the string in the collection of gzipped training data of that network (if no utility allows you to do that you can at least unzip all .gz files). Since the hexadecimal string encodes the position, it's very unlikely that you find two games sharing a string in the late game.

**Comment by Ishinoshita** [comment]
@alreadydone I see the 'trick' and it would indeed be straightforwards. But as I understand it,  dump_supervised does not come as an .exe with std LZ + Autogtp package. I would need to dowload and compile Training.cpp and all its dependencies. That's to much on the plate of a beginner like me.

Thank you anyway. Since I don't plan move on cpp ground for time being, I think I will try a workaround with python only, namely modify the sgf to ascii sript of sgfmill library to genrerate a string representation, then convert it to hex representation.

**Comment by alreadydone** [comment]
No, dump_supervised is implemented as a GTP command and you can just open up leelaz in command line and type in the command. I tried to convert a new self-play game and [1e82.0.gz](https://github.com/gcp/leela-zero/files/2521972/1e82.0.gz) is the result; it's not actual training data as probabilities are concentrated at a single move, but it has all the positions.

**Comment by Ishinoshita** [comment]
Thanks @alreadydone! It works very well, indeed. Avoids me the pain to dig into sgfmill code (although one of my goal is to learn python ;-).

I have back-gererated supersived training data from one of the sgf my script had generated  from LZ181 training data (thus, I perfectly knew in which chunk/game the hex string was to be found). And Notepad++ retrieved the correct chunk/game (although it took ~1s/chunk on my PC, so could take quite a while for thousands of chunks, the typical size of one training .zip).

*To have dump_supervised accept train2sgf output format, I just had to strip the sgf from all the comments that this script embed after each move (policy data in the position reached after the move) and to add the result of the game (something I missed to include in my script and that I will add in the future).*
